cmd/ci-secret-bootstrap/main_test.go

```diff
@@ -0,0 +1,1148 @@
+			given: options{
+				bwPassword: "topSecret",
+				config: Config{
+					SecretConfig{
+						From: map[string]BWContext{
+							"key-name-1": {
+								BWItem: "item-name-1",
+								Field:  "field-name-1",
+							},
+						},
+						To: []SecretContext{
+							{
+								Cluster:   "default",
+								Namespace: "namespace-1",
+								Name:      "prod-secret-1",
+							},
+							{
+								Cluster:   "build01",
+								Namespace: "namespace-1",
+								Name:      "prod-secret-1",
+							},
+							{
+								Cluster:   "default",
+								Namespace: "namespace-1",
+								Name:      "prod-secret-1",
+							},
+						},
+					},
+				},
+				kubeConfigs: map[string]rest.Config{
+					"default": {
+						Host:        "https://api.ci.openshift.org:443",
+						BearerToken: "token1",
+					},
+					"build01": {
+						Host:        "https://api.build01.ci.devcluster.openshift.com:6443",
+						BearerToken: "token2",
+					},
+				},
+			},
+			expected: fmt.Errorf("secret context '{default namespace-1 prod-secret-1}' showed more than once in the config"),
+		},
+		{
+			name: "conflicting secrets in different TOs",
+			given: options{
+				bwPassword: "topSecret",
+				config: Config{
+					SecretConfig{
+						From: map[string]BWContext{
+							"key-name-1": {
+								BWItem: "item-name-1",
+								Field:  "field-name-1",
+							},
+						},
+						To: []SecretContext{
+							{
+								Cluster:   "build01",
+								Namespace: "namespace-1",
+								Name:      "prod-secret-1",
+							},
+							{
+								Cluster:   "default",
+								Namespace: "namespace-1",
+								Name:      "prod-secret-1",
+							},
+						},
+					},
+					SecretConfig{
+						From: map[string]BWContext{
+							"key-name-1": {
+								BWItem: "item-name-1",
+								Field:  "field-name-1",
+							},
+						},
+						To: []SecretContext{
+							{
+								Cluster:   "default",
+								Namespace: "namespace-1",
+								Name:      "prod-secret-1",
+							},
+							{
+								Cluster:   "build01",
+								Namespace: "namespace-1",
+								Name:      "prod-secret-1",
+							},
+						},
+					},
+				},
+				kubeConfigs: map[string]rest.Config{
+					"default": {
+						Host:        "https://api.ci.openshift.org:443",
+						BearerToken: "token1",
+					},
+					"build01": {
+						Host:        "https://api.build01.ci.devcluster.openshift.com:6443",
+						BearerToken: "token2",
+					},
+				},
+			},
+			expected: fmt.Errorf("secret context '{default namespace-1 prod-secret-1}' showed more than once in the config"),
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			actual := tc.given.validateCompletedOptions()
+			if tc.expected == nil {
+				assert.Nil(t, actual)
+			} else {
+				assert.EqualError(t, actual, tc.expected.Error())
+			}
+		})
+	}
+}
+
+func getGetFakeSecretGetter(clients map[string]*fake.Clientset) func(map[string]rest.Config, string) (coreclientset.SecretsGetter, error) {
```

Comment:
Couldn't you just set a different set of `clients`?

Reply:
The actual logic for choosing which client should not be mocked out -- this code will not test, for instance, that the production code actually chooses the correct client to use for each secret